vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/codec/json/JsonObjectCodec.java

```diff
@@ -106,6 +149,14 @@ protected Object readObjectId(BsonReader reader, DecoderContext ctx) {
 
   @Override
   protected Object readDateTime(BsonReader reader, DecoderContext ctx) {
```

Comment:
According to the mongo docs a long should be used if the time is before the epoch (as you can't then format a sensible ISO string):

See "data_date" here http://docs.mongodb.org/manual/reference/mongodb-extended-json/

I'm thinking something along the lines of this (this is what I hacked earlier but I don't think it quite works) :

https://gist.github.com/purplefox/8d9ba91ff4c20e28c393